src/zjs_ocf_ble.c

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2016-2017, Intel Corporation.
+
+#include "zjs_ocf_ble.h"
+
+static bt_addr_le_t id_addr;
+
+static ssize_t zjs_ble_storage_read(const bt_addr_le_t *addr, uint16_t key,
+    void *data, size_t length)
+{
+    if (addr) {
+        return -ENOENT;
+    }
+
+    if (key == BT_STORAGE_ID_ADDR && length == sizeof(id_addr) &&
+        bt_addr_le_cmp(&id_addr, BT_ADDR_LE_ANY)) {
+        bt_addr_le_copy(data, &id_addr);
+        return sizeof(id_addr);
+    }
+
+    return -EIO;
+}
+
+static int char2hex(const char *c, uint8_t *x)
```

Comment:
The input is hex, not the output, so maybe hex2int? Seems better to me to take char c and maybe just return uint8_t, 0xff could be failure?

Just less tricky looking dropping the pointers.

Oh... I already have a zjs_hex_to_byte in zjs_util.h that is way too similar. Don't have to do this now but let's make a note to self to consolidate these implementations. I also have something doing the reverse in zjs_buffer.c I think.